server/src/main/java/org/apache/seata/server/lock/impl/CGLockManagerImpl.java

```diff
@@ -0,0 +1,221 @@
+package org.apache.seata.server.lock.impl;
+
+import org.apache.seata.common.exception.StoreException;
+import org.apache.seata.common.util.CollectionUtils;
+import org.apache.seata.core.exception.BranchTransactionException;
+import org.apache.seata.core.lock.RowLock;
+import org.apache.seata.server.lock.CGLockManager;
+import org.apache.seata.server.lock.dto.CGLockRequest;
+import org.apache.seata.server.lock.dto.LockResult;
+import org.apache.seata.server.lock.holder.ObjectHolder;
+import org.apache.seata.server.lock.storage.BucketLockMap;
+import org.apache.seata.server.lock.storage.CGLockStorage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+import static org.apache.seata.core.exception.TransactionExceptionCode.LockKeyConflictFailFast;
+
+/**
+ * Default implementation of CGLockManager for distributed lock management
+ */
+@Service
+public class CGLockManagerImpl implements CGLockManager {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CGLockManagerImpl.class);
+
+    // Track lock holders by ObjectHolder for efficient cleanup
+    private final Map<ObjectHolder, Set<String>> lockHolderMap = new ConcurrentHashMap<>();
```

Comment:
I'm not sure why we need lockHolderMap when CGLockStorage already contains ObjectHolder.